src/main/client/scene/game-scene.ts

```diff
@@ -52,22 +60,40 @@ export class GameScene extends Scene {
          terrain.closePath();
          terrain.fillPath();
       }
-      terrain.generateTexture('terrain', sceneWidth, sceneHeight);
-      terrain.clear();
-      this.add.sprite(sceneWidth / 2, sceneHeight / 2, 'terrain');
+      this.terrainTexture = this.textures.createCanvas('terrain', sceneWidth, sceneHeight);
+      terrain.generateTexture(this.terrainTexture.getCanvas(), sceneWidth, sceneHeight);
+      this.add.sprite(0, 0, 'terrain');
    }
 
-   private hitTestTerrain(x: number, y: number, w: number, h: number): boolean {
-      for (let i = x; i < x + w; i++) {
-         for (let j = y; j < y + h; j++) {
-            if (this.textures.getPixelAlpha(i, j, 'terrain')) {
+   private hitTestTerrain(worldX: number, worldY: number, width: number, height: number): boolean {
+      const data = this.terrainTexture.getData(worldX, worldY, this.characterSize, this.characterSize2);
+      for (let i = 0; i < width; i++) {
+         for (let j = 0; j < height; j++) {
+            const pixel = this.getPixel(i, j, data);
+            if (pixel && pixel.green > 0) {
                return true;
             }
          }
       }
       return false;
    }
 
+   private getPixel(localX: number, localY: number, canvasData: ImageData): Color {
+      if (localX < 0 || localY < 0 || localX > canvasData.width || localY > canvasData.height) return;
+
+      const r = (localY * canvasData.width + localX) * 4;
+      const g = (localY * canvasData.width + localX) * 4 + 1;
+      const b = (localY * canvasData.width + localX) * 4 + 2;
+      const a = (localY * canvasData.width + localX) * 4 + 3;
+
+      return {
```

Comment:
We should return only the alpha value instead of creating an object every time.

Reply:
will do

Reply:
actually, the method is called "getPixel", so it should return a color object

Reply:
i improved the method now